designer/server/src/main/scala/pl/touk/nussknacker/ui/process/periodic/PeriodicDeploymentManagerDecorator.scala

```diff
@@ -0,0 +1,74 @@
+package pl.touk.nussknacker.ui.process.periodic
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.component.ScenarioPropertyConfig
+import pl.touk.nussknacker.engine.api.definition.{MandatoryParameterValidator, StringParameterEditor}
+import pl.touk.nussknacker.engine.api.deployment.periodic.services.{
+  EmptyPeriodicProcessListenerFactory,
+  PeriodicSchedulePropertyExtractorFactory,
+  ProcessConfigEnricherFactory
+}
+import pl.touk.nussknacker.engine.api.deployment.{DeploymentManager, PeriodicExecutionSupported}
+import pl.touk.nussknacker.engine.{DeploymentManagerDependencies, ModelData}
+import pl.touk.nussknacker.ui.process.periodic.cron.{CronParameterValidator, CronSchedulePropertyExtractor}
+
+object PeriodicDeploymentManagerDecorator extends LazyLogging {
+
+  def decorate(
+      underlying: DeploymentManager,
+      periodicExecutionSupported: PeriodicExecutionSupported,
+      periodicProcessesManagerProvider: PeriodicProcessesManagerProvider,
+      modelData: ModelData,
+      deploymentConfig: Config,
+      dependencies: DeploymentManagerDependencies,
+  ): DeploymentManager = {
+    logger.info("Decorating DM with periodic functionality")
+    import net.ceedubs.ficus.Ficus._
+    import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+    val periodicBatchConfig = deploymentConfig.as[PeriodicBatchConfig]("deploymentManager")
```

Comment:
Can we have all scheduling related config in one field? E.g.:
```
scheduling {
  enabled: true
  ... other configuration options
}
```
? 
`deploymentManager` looks weird here as all `deploymentConfig` is a configuration of a deployment manager.

Reply:
Done, new config section, also renamed class in code PeriodicBatchConfig -> SchedulingConfig.